src/c++/perf_analyzer/load_manager.cc

```diff
@@ -143,6 +143,20 @@ LoadManager::ResetIdleTime()
   }
 }
 
+const size_t
+LoadManager::GetAndResetNumSentRequests()
+{
+  size_t num_sent_requests{0};
+
+  for (auto& thread_stat : threads_stat_) {
+    std::lock_guard<std::mutex> lock(thread_stat->mu_);
+    num_sent_requests += thread_stat->num_sent_requests_;
```

Comment:
Can we use atomic variable instead of mutex to avoid blocking all the threads from making progress while the reporter thread fetches this number?

Reply:
This will not block all threads. It will only block the one thread it is reading from.
I'm not against turning these into atomics though. Do atomics work out of the box? Or do you have to do an explicit atomic load?

Reply:
Switched to atomic 👍